{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Paper:\tGoldman JA, Kuzu G, Lee N, Karasik J et al. Resolving Heart Regeneration by Replacement Histone Profiling. Dev Cell 2017 Feb 27;40(4):392-404.e5. PMID: 28245924  \n",
    "\n",
    "Data Title: \tGenome-wide maps of histone variant H3.3 occupancy in zebrafish cardiomyocytes, Danio rerio  \n",
    "Platform: Illumina HiSeq 4000 (Danio rerio)  \n",
    "GEO Series:  GSE81893  \n",
    "\n",
    "Data Processing  \n",
    "ChIP-seq reads were aligned to the GRCz10 zebrafish genome assembly using Bowtie with parameters “-M 5 -k 1 -I 50 -X 500 --solexa-quals –best”\n",
    "The positions with anomalously high read counts are potential amplification artifacts and were removed from consideration Peaks in the ChIP-Seq fragments distribution were identified using the SPP software package with default parameters  \n",
    "Genome_build: GRCz10 zebrafish genome\n",
    "Supplementary_files_format_and_content: peak files were generated using spp; scores show the enrichment  \n",
    "\n",
    "Organism:\tDanio rerio  \n",
    "\n",
    "Characteristics\texpression: transgenic\n",
    "genotype: (cmlc2:H3.3-bio)pd185\n",
    "chromatin pulldown: streptavidin  \n",
    "\n",
    "treatment: .4 uM tamoxifen 18 hrs  \n",
    "\n",
    "time: 7-24 days post incubation  \n",
    "cell type: cardiomyocyte  \n",
    "Treatment protocol\tFish were incubated in .4 uM tamoxifen for 18 hrs. Hearts were dissected 14 days later.  \n",
    "Growth protocol\tWild-type or transgenic male and female zebrafish of the outbred Ekkwill (EK) strain were used for all experiments, with adults ranging in age from 4 to 6 months. Water temperature was maintained at 26°C for animals .  \n",
    "Extracted molecule\tgenomic DNA  \n",
    "Extraction protocol\tWhole ventricles from 20-30 fish were grinded up and nuclei were extracted using digitonin. Chromatin was then solubilized using micrococcal nuclease.  \n",
    "\n",
    "Libraries were preapred as described in Bowman et al. BMC Genomics 14:466 (2013)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat >config_goldmanH33_data.yaml<<EOL\n",
    "\n",
    "samples:\n",
    "  Uninjured_input: \"SRX1799048\"\n",
    "  Uninjured_chip_rep1: \"SRX1799049\"\n",
    "  Uninjured_chip_rep2: \"SRX1799050\"\n",
    "  Uninjured_chip_rep3: \"SRX1799051\"\n",
    "  14dpi_input: \"SRX1799052\"\n",
    "  14dpi_chip_rep1: \"SRX1799053\"\n",
    "  14dpi_chip_rep2: \"SRX1799054\"\n",
    "  7dpt_input: \"SRX1799055\"\n",
    "  7dpt_chip_rep1: \"SRX1799056\"\n",
    "  7dpt_chip_rep2: \"SRX1799057\"\n",
    "  \n",
    "fastq_path:\n",
    "    \"/fast/AG_Ohler/Vic/0_Projects/Goldman_injured/data/\"\n",
    "    \n",
    "reads:\n",
    "- 1\n",
    "\n",
    "replicas:\n",
    "- 1\n",
    "- 2\n",
    "\n",
    "bigWigs:\n",
    "- Uninjured_chip_rep1\n",
    "- Uninjured_chip_rep2\n",
    "- 14dpi_chip_rep1\n",
    "- 14dpi_chip_rep2\n",
    "- 7dpt_chip_rep1\n",
    "- 7dpt_chip_rep2\n",
    "\n",
    "EOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import snakemake.io\n",
    "from snakemake.io import expand\n",
    "from snakemake.io import glob_wildcards\n",
    "import yaml\n",
    "import itertools\n",
    "\n",
    "with open(\"config_goldmanH33_data.yaml\", \"r\") as config_file:\n",
    "        config = yaml.safe_load(config_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat>script_download_sra_files.py<<EOL\n",
    "\n",
    "import yaml\n",
    "import subprocess\n",
    "\n",
    "print(\"let's start\")\n",
    "with open(\"config_goldmanH33_data.yaml\", \"r\") as config_file:\n",
    "        config = yaml.safe_load(config_file)\n",
    "\n",
    "sra_files = config['samples'] \n",
    "\n",
    "for name, number in sra_files.items():\n",
    "    print(\"Downloading\", number)\n",
    "    subprocess.call('fastq-dump --gzip --split-files $number --outdir ../data/', shell=True)\n",
    "    subprocess.call('mv $number.fastq.gz $name.fastq.gz', shell=True)\n",
    "    \n",
    "EOL\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../src'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-57-d9c6c26e3847>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"../src\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"config_goldmanH33_data.yaml\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"r\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mconfig_file\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m         \u001b[0mconfig\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0myaml\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msafe_load\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconfig_file\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../src'"
     ]
    }
   ],
   "source": [
    "import os \n",
    "import yaml\n",
    "import subprocess\n",
    "\n",
    "os.chdir(\"../src\")\n",
    "with open(\"config_goldmanH33_data.yaml\", \"r\") as config_file:\n",
    "        config = yaml.safe_load(config_file)\n",
    "\n",
    "sra_files = config['samples'] \n",
    "\n",
    "\n",
    "#os.chdir(\"../data\")\n",
    "os.chdir('../results/fastqc/')\n",
    "for name, number in sra_files.items():\n",
    "    print(\"renaming\", number)\n",
    "    #subprocess.call(['mv {}*.fastq.gz {}.fastq.gz'.format(number,name)], shell=True)\n",
    "    #subprocess.call(['mv {}_1.fastq.gz {}.fastq.gz '.format(name,name)], shell=True)\n",
    "    subprocess.call(['mv {}_1.fastqc.html {}.fastqc.html '.format(name,name)], shell=True)\n",
    "    print(\"done\")\n",
    "    \n",
    "os.chdir(\"../src\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os \n",
    "import yaml\n",
    "import subprocess\n",
    "\n",
    "os.chdir(\"../data\")\n",
    "subprocess.call('mv SRX1799048_1.fastq.gz name.fastq.gz', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import yaml\n",
    "import subprocess\n",
    "\n",
    "with open(\"config_goldmanH33_data.yaml\", \"r\") as config_file:\n",
    "        config = yaml.safe_load(config_file)\n",
    "\n",
    "sra_files = config['samples'] \n",
    "\n",
    "for name, number in sra_files.items():\n",
    "    subprocess.call('echo this is $number', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#qsub file\n",
    "cat>qsub_goldmanH33.py<<EOL\n",
    "#!/usr/bin/python\n",
    "#$ -N goldmanH33    # name of the job\n",
    "#$ -m ea                   # send an email when end or job aborted \n",
    "#$ -M Vic-Fabienne.Schumann@mdc-berlin.de \n",
    "#$ -wd   /fast/AG_Ohler/Vic/0_Projects/Goldman_injured/src    # working dir\n",
    "#$ -j y                        # combine error with stdout\n",
    "#$ -o log_sradownload.stdlog      # default filename for stdout and error file \n",
    "#$ -l h_vmem=20G    # limit memory request\n",
    "#$ -l h_stack=128M    # often not needed, but a few programs require it\n",
    "#$ -pe smp 5            # smp parallel environment\n",
    "\n",
    "\n",
    "\n",
    "# I have this in every qsub file, so that I know what the job was, what node it was on if there are problems, and the resources I requested\n",
    "\n",
    "\n",
    "# BODY: run the program of interest \n",
    " # source your guix profile to have access to all your tools\n",
    "\n",
    "import yaml\n",
    "import subprocess\n",
    "\n",
    "print(\"let's start\")\n",
    "with open(\"config_goldmanH33_data.yaml\", \"r\") as config_file:\n",
    "        config = yaml.safe_load(config_file)\n",
    "\n",
    "sra_files = config['samples'] \n",
    "\n",
    "for name, number in sra_files.items():\n",
    "    print(\"Downloading\", number)\n",
    "    subprocess.call(['$HOME/.guix-profile/bin/fastq-dump --gzip --split-files  --outdir ../data/ {}'.format(number)], shell=True)\n",
    "    #for single-end\n",
    "    subprocess.call(['mv {}_1.fastq.gz {}.fastq.gz'.format(number,name)], shell=True)\n",
    "    # for paired-end\n",
    "    #subprocess.call(['mv {}_1.fastq.gz {}_1.fastq.gz'.format(number,name)], shell=True)\n",
    "    #subprocess.call(['mv {}_2.fastq.gz {}_2.fastq.gz'.format(number,name)], shell=True)\n",
    "\n",
    "# at the bottom of every qsub file I have the below so that I know the resources that were used for that particular job. I do get this in the final email, but sometimes it's nicer to have it searchable in the log file.\n",
    "\n",
    "EOL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Snakemake for Goldman data analysis\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat >rules/fastqc_singleend.smk<<EOL\n",
    "\n",
    "rule fastq:\n",
    "    input: \n",
    "        \"../data/{sample}.fastq.gz\"\n",
    "    output: \n",
    "        \"fastqc/{sample}.html\"\n",
    "    shell:\n",
    "        \"fastqc -o 'fastqc/ {output}' {input}\"\n",
    "EOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd ../results/fastqc/\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 1: cd: ../src/: No such file or directory\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'cd ../src/\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-61-782b1b4516fd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'cd ../src/\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.guix-profile/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2350\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2351\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2352\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2353\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2354\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.guix-profile/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/.guix-profile/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.guix-profile/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'cd ../src/\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ../src/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat >rules/bowtie2_chip_without_ctdpt.smk<<EOL\n",
    "\n",
    "bowtie2_index = \"/fast/AG_Ohler/Scott/danRer11/noAlt_danRer11\"\n",
    "\n",
    "rule aligning_bowtie2:\n",
    "        input:\n",
    "            \"../data/{sample}.fastq.gz\"\n",
    "        output:\n",
    "            sam= \"mapped/mapped_{sample}.sam\",\n",
    "            log= \"mapped/mapped_{sample}.bwt2.log\"\n",
    "        params:\n",
    "            index=bowtie2_index\n",
    "        shell:\n",
    "            \" bowtie2  --sensitive -x {params.index} -U  {input} -p 4 -S {output.sam} 2> {output.log}\"\n",
    "\n",
    "EOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat >rules/samtools_singleend.smk<<EOL\n",
    "\n",
    "rule samtools:\n",
    "    input:\n",
    "        \"mapped/mapped_{sample}.sam\"\n",
    "    output:\n",
    "        \"samtools/sammark_mapped_{sample}.bam\"\n",
    "    threads: 4\n",
    "    shell:\n",
    "        \" samtools view -@ {threads} -q 30 -uSbF 4 {input}| samtools sort -l 0 - | \\\n",
    "        samtools markdup -r -s - - | \\\n",
    "        samtools sort -l 0 - > {output}\"\n",
    "        \n",
    "rule indexing_samtools:\n",
    "        input:\n",
    "              \"samtools/sammark_mapped_{sample}.bam\"\n",
    "        output:\n",
    "             \"samtools/sammark_mapped_{sample}.bam.bai\" \n",
    "        shell:\n",
    "             \"samtools index {input} {output}\"\n",
    "\n",
    "EOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_names = config['samples']\n",
    "sample_names = sample_names.keys()\n",
    "list(sample_names)\n",
    "\n",
    "list_of_base = []\n",
    "for sample in list(sample_names):\n",
    "    base = (sample.split('_')[0])\n",
    "    if base not in list_of_base:\n",
    "        list_of_base.append(base)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>col1</th>\n",
       "      <th>col2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Uninjured</td>\n",
       "      <td>input</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Uninjured</td>\n",
       "      <td>chip_rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Uninjured</td>\n",
       "      <td>chip_rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Uninjured</td>\n",
       "      <td>chip_rep3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14dpi</td>\n",
       "      <td>input</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>14dpi</td>\n",
       "      <td>chip_rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>14dpi</td>\n",
       "      <td>chip_rep2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7dpt</td>\n",
       "      <td>input</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7dpt</td>\n",
       "      <td>chip_rep1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>7dpt</td>\n",
       "      <td>chip_rep2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        col1       col2\n",
       "0  Uninjured      input\n",
       "1  Uninjured  chip_rep1\n",
       "2  Uninjured  chip_rep2\n",
       "3  Uninjured  chip_rep3\n",
       "4      14dpi      input\n",
       "5      14dpi  chip_rep1\n",
       "6      14dpi  chip_rep2\n",
       "7       7dpt      input\n",
       "8       7dpt  chip_rep1\n",
       "9       7dpt  chip_rep2"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "sample_names = config['samples']\n",
    "sample_names = sample_names.keys()\n",
    "basis = []\n",
    "tails = []\n",
    "dict_sample_parts = {}\n",
    "for sample in list(sample_names):\n",
    "    basis.append(sample.split('_')[0])\n",
    "    tails.append(sample.split('_',1)[1])\n",
    "    \n",
    "\n",
    "data = {'col1':basis, 'col2':tails} \n",
    "df = pd.DataFrame(data=data, columns=None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {},
   "outputs": [],
   "source": [
    "from snakemake.io import expand\n",
    "\n",
    "import yaml\n",
    "import json\n",
    "\n",
    "yaml_text = \"\"\"\n",
    "samples:\n",
    "   base: \n",
    "    - Uninjured: \n",
    "        replica: [chip_rep1, chip_rep2, chip_rep3]\n",
    "    - 14dpi:\n",
    "        replica: [chip_rep1,chip_rep2]\n",
    "        \n",
    "files:\n",
    "    data:\n",
    "    - Uninjured_H3K27Ac_ChIP\n",
    "    - heart_1year_H3K27ac_ChIP-seq_R1\n",
    "    - heart_1year_H3K27ac_ChIP-seq_R2\n",
    "    control:\n",
    "    - Uninjured_input \n",
    "    - heart_1year_input_DNA_R1\n",
    "    - heart_1year_input_DNA_R2\n",
    "    \n",
    "stupid:\n",
    "    Uninjured:\n",
    "      - chip_rep1\n",
    "      - chip_rep2\n",
    "      - chip_rep3\n",
    "    14dpi:\n",
    "      - chip_rep1\n",
    "      - chip_rep2\n",
    "\"\"\"  \n",
    "yaml = yaml.load(yaml_text)\n",
    "\n",
    "\n",
    "json_text = \"\"\"\n",
    "{\n",
    "\"Uninjured\":{\n",
    "    \"base\":\"Uninjured\",\n",
    "    \"replicas\":[\"chip_rep1\",\"chip_rep2\",\"chip_rep3\"]\n",
    "},\n",
    "\"14dpi\":{\n",
    "    \"base\":\"14dpi\",\n",
    "    \"replicas\":[\"chip_rep1\",\"chip_rep2\"]\n",
    "}\n",
    "}\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['Uninjured', '14dpi'])"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(json_text)\n",
    "loaded_json = (json.loads(json_text))\n",
    "loaded_json.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'list' object has no attribute 'split'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-268-cacd993811c0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mreplicate\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0myaml\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"stupid\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Uninjured\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"_\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mreplicate\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'list' object has no attribute 'split'"
     ]
    }
   ],
   "source": [
    "replicate = yaml[\"stupid\"][\"Uninjured\"]\n",
    "replicate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-265-fa1ef9421b70>, line 8)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-265-fa1ef9421b70>\"\u001b[0;36m, line \u001b[0;32m8\u001b[0m\n\u001b[0;31m    print(other_input_list)\u001b[0m\n\u001b[0m        ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "#basis = [s[\"base\"] for s in yaml[\"samples\"]]\n",
    "#print(basis)\n",
    "\n",
    "input_list = expand(\"{samples}.bam\", samples=yaml[\"samples\"][\"base\"])\n",
    "print(input_list)\n",
    "\n",
    "other_input_list = expand(\"{samples}_{replicate}.bam\", samples=yaml[\"stupid\"], replicate = yaml[\"stupid\"][\"Uninjured\"]\n",
    "print(other_input_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Uninjured_input', '14dpi_input']"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "control_files = [s[\"control\"] for s in yaml['samples']]\n",
    "control_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat >rules/macs2_Chip_se.smk<<EOL\n",
    "\n",
    "\n",
    "rule macs2:\n",
    "    input:\n",
    "        data= \"samtools/sammark_mapped_{base}_chip_rep{replicate}.bam\",\n",
    "        control= \"samtools/sammark_mapped_{base}_input.bam\"\n",
    "    output:\n",
    "        \"macs/{base}_chip_rep{replicate}_peaks.narrowPeak\"\n",
    "    shell:\n",
    "        \"macs2 callpeak -t {input.data} -c {input.control} -g 1.6e+9  -f BAM --keep-dup all --outdir macs -n {wildcards.base}_chip_rep{wildcards.replicate} -B  -q 0.01\" \n",
    "\n",
    "EOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat >snakemake_goldmanH33_data.py<<EOL\n",
    "\n",
    "configfile: '../src/config_goldmanH33_data.yaml'\n",
    "\n",
    "FILE_NAME  = config[\"samples\"]\n",
    "\n",
    "#For macs2 \n",
    "sample_names = config['samples']\n",
    "sample_names = sample_names.keys()\n",
    "\n",
    "list_of_base = []\n",
    "for sample in list(sample_names):\n",
    "    base = (sample.split('_')[0])\n",
    "    if base not in list_of_base:\n",
    "        list_of_base.append(base)\n",
    "        \n",
    "merged_sample_names = []\n",
    "for samplename in FILE_NAME:\n",
    "    if 'wt' in samplename:\n",
    "        if 'rep1' in samplename:\n",
    "            merged_sample_names.append(samplename[:-5])\n",
    "    \n",
    "\n",
    "\n",
    "rule all:\n",
    "    input:\n",
    "##fastqc\n",
    "#        expand(\"fastqc/{sample}.html\", sample=list(FILE_NAME.keys())), #from subexp dir to results/\n",
    "#no cutadapt needed here\n",
    "###bowtie2\n",
    "#        expand(\"mapped/mapped_{sample}.sam\", sample=list(FILE_NAME.keys())),\n",
    "#        expand(\"mapped/mapped_{sample}.bwt2.log\", sample=list(FILE_NAME.keys())),\n",
    "#samtools\n",
    "        expand(\"samtools/sammark_mapped_{sample}.bam\", sample=list(FILE_NAME.keys())),\n",
    "        expand(\"samtools/sammark_mapped_{sample}.bam.bai\", sample=list(FILE_NAME.keys())),\n",
    "#macs\n",
    "        expand(\"macs/{base}_chip_rep{replicate}_peaks.narrowPeak\", base=list_of_base, replicate=config['replicas']),\n",
    "#bigWigs\n",
    "        expand(\"bigWigs/{sample}_treat_pileup_sort.bigWig\", sample=config[\"bigWigs\"]),\n",
    "        expand(\"bigWigs/{sample}_control_lambda_sort.bigWig\", sample=config[\"bigWigs\"]),\n",
    "        expand(\"bigWigs/{sample}_subtract_sort.bigWig\", sample=config[\"bigWigs\"])\n",
    "        \n",
    "#include: \"rules/fastqc_singleend.smk\"\n",
    "include: \"rules/bowtie2_chip_without_ctdpt.smk\"\n",
    "include: \"rules/samtools_singleend.smk\"\n",
    "include: \"rules/macs2_Chip_se.smk\"\n",
    "include: \"rules/generatingBigWig-Copy1.smk\"\n",
    "\n",
    "EOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Job counts:\n",
      "\tcount\tjobs\n",
      "\t1\tall\n",
      "\t6\tbdg2bigWig\n",
      "\t6\tbdg2bigWigCtl\n",
      "\t6\tbdgSort\n",
      "\t6\tbdgSortCtl\n",
      "\t6\tbdgcomp\n",
      "\t6\tsub2bigWig\n",
      "\t6\tsubSort\n",
      "\t43\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSort:\n",
      "    input: macs/Uninjured_chip_rep1_treat_pileup.bdg\n",
      "    output: bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bedGraph\n",
      "    jobid: 62\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "bedtools sort -i macs/Uninjured_chip_rep1_treat_pileup.bdg > bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSortCtl:\n",
      "    input: macs/7dpt_chip_rep1_control_lambda.bdg\n",
      "    output: bigWigs/7dpt_chip_rep1_control_lambda_sort.bedGraph\n",
      "    jobid: 64\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "bedtools sort -i macs/7dpt_chip_rep1_control_lambda.bdg > bigWigs/7dpt_chip_rep1_control_lambda_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSortCtl:\n",
      "    input: macs/14dpi_chip_rep2_control_lambda.bdg\n",
      "    output: bigWigs/14dpi_chip_rep2_control_lambda_sort.bedGraph\n",
      "    jobid: 57\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "bedtools sort -i macs/14dpi_chip_rep2_control_lambda.bdg > bigWigs/14dpi_chip_rep2_control_lambda_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSort:\n",
      "    input: macs/Uninjured_chip_rep2_treat_pileup.bdg\n",
      "    output: bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bedGraph\n",
      "    jobid: 66\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "bedtools sort -i macs/Uninjured_chip_rep2_treat_pileup.bdg > bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSort:\n",
      "    input: macs/14dpi_chip_rep1_treat_pileup.bdg\n",
      "    output: bigWigs/14dpi_chip_rep1_treat_pileup_sort.bedGraph\n",
      "    jobid: 54\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "bedtools sort -i macs/14dpi_chip_rep1_treat_pileup.bdg > bigWigs/14dpi_chip_rep1_treat_pileup_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSortCtl:\n",
      "    input: macs/7dpt_chip_rep2_control_lambda.bdg\n",
      "    output: bigWigs/7dpt_chip_rep2_control_lambda_sort.bedGraph\n",
      "    jobid: 65\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "bedtools sort -i macs/7dpt_chip_rep2_control_lambda.bdg > bigWigs/7dpt_chip_rep2_control_lambda_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSortCtl:\n",
      "    input: macs/Uninjured_chip_rep2_control_lambda.bdg\n",
      "    output: bigWigs/Uninjured_chip_rep2_control_lambda_sort.bedGraph\n",
      "    jobid: 60\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "bedtools sort -i macs/Uninjured_chip_rep2_control_lambda.bdg > bigWigs/Uninjured_chip_rep2_control_lambda_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSort:\n",
      "    input: macs/14dpi_chip_rep2_treat_pileup.bdg\n",
      "    output: bigWigs/14dpi_chip_rep2_treat_pileup_sort.bedGraph\n",
      "    jobid: 67\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "bedtools sort -i macs/14dpi_chip_rep2_treat_pileup.bdg > bigWigs/14dpi_chip_rep2_treat_pileup_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSort:\n",
      "    input: macs/7dpt_chip_rep1_treat_pileup.bdg\n",
      "    output: bigWigs/7dpt_chip_rep1_treat_pileup_sort.bedGraph\n",
      "    jobid: 52\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "bedtools sort -i macs/7dpt_chip_rep1_treat_pileup.bdg > bigWigs/7dpt_chip_rep1_treat_pileup_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSort:\n",
      "    input: macs/7dpt_chip_rep2_treat_pileup.bdg\n",
      "    output: bigWigs/7dpt_chip_rep2_treat_pileup_sort.bedGraph\n",
      "    jobid: 58\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "bedtools sort -i macs/7dpt_chip_rep2_treat_pileup.bdg > bigWigs/7dpt_chip_rep2_treat_pileup_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSortCtl:\n",
      "    input: macs/14dpi_chip_rep1_control_lambda.bdg\n",
      "    output: bigWigs/14dpi_chip_rep1_control_lambda_sort.bedGraph\n",
      "    jobid: 70\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "bedtools sort -i macs/14dpi_chip_rep1_control_lambda.bdg > bigWigs/14dpi_chip_rep1_control_lambda_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgSortCtl:\n",
      "    input: macs/Uninjured_chip_rep1_control_lambda.bdg\n",
      "    output: bigWigs/Uninjured_chip_rep1_control_lambda_sort.bedGraph\n",
      "    jobid: 49\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "bedtools sort -i macs/Uninjured_chip_rep1_control_lambda.bdg > bigWigs/Uninjured_chip_rep1_control_lambda_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWigCtl:\n",
      "    input: bigWigs/Uninjured_chip_rep2_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep2_control_lambda_sort.bigWig\n",
      "    jobid: 24\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/Uninjured_chip_rep2_control_lambda_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/Uninjured_chip_rep2_control_lambda_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgcomp:\n",
      "    input: bigWigs/14dpi_chip_rep1_treat_pileup_sort.bedGraph, bigWigs/14dpi_chip_rep1_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep1_subtract.bedGraph\n",
      "    jobid: 73\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "macs2 bdgcmp -t bigWigs/14dpi_chip_rep1_treat_pileup_sort.bedGraph -c bigWigs/14dpi_chip_rep1_control_lambda_sort.bedGraph -m subtract -o bigWigs/14dpi_chip_rep1_subtract.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWig:\n",
      "    input: bigWigs/14dpi_chip_rep1_treat_pileup_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep1_treat_pileup_sort.bigWig\n",
      "    jobid: 15\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/14dpi_chip_rep1_treat_pileup_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/14dpi_chip_rep1_treat_pileup_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWigCtl:\n",
      "    input: bigWigs/Uninjured_chip_rep1_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep1_control_lambda_sort.bigWig\n",
      "    jobid: 9\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/Uninjured_chip_rep1_control_lambda_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/Uninjured_chip_rep1_control_lambda_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgcomp:\n",
      "    input: bigWigs/7dpt_chip_rep1_treat_pileup_sort.bedGraph, bigWigs/7dpt_chip_rep1_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep1_subtract.bedGraph\n",
      "    jobid: 77\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "macs2 bdgcmp -t bigWigs/7dpt_chip_rep1_treat_pileup_sort.bedGraph -c bigWigs/7dpt_chip_rep1_control_lambda_sort.bedGraph -m subtract -o bigWigs/7dpt_chip_rep1_subtract.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWig:\n",
      "    input: bigWigs/14dpi_chip_rep2_treat_pileup_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep2_treat_pileup_sort.bigWig\n",
      "    jobid: 36\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/14dpi_chip_rep2_treat_pileup_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/14dpi_chip_rep2_treat_pileup_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgcomp:\n",
      "    input: bigWigs/14dpi_chip_rep2_treat_pileup_sort.bedGraph, bigWigs/14dpi_chip_rep2_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep2_subtract.bedGraph\n",
      "    jobid: 76\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "macs2 bdgcmp -t bigWigs/14dpi_chip_rep2_treat_pileup_sort.bedGraph -c bigWigs/14dpi_chip_rep2_control_lambda_sort.bedGraph -m subtract -o bigWigs/14dpi_chip_rep2_subtract.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWigCtl:\n",
      "    input: bigWigs/7dpt_chip_rep2_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep2_control_lambda_sort.bigWig\n",
      "    jobid: 32\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/7dpt_chip_rep2_control_lambda_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/7dpt_chip_rep2_control_lambda_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWig:\n",
      "    input: bigWigs/7dpt_chip_rep1_treat_pileup_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep1_treat_pileup_sort.bigWig\n",
      "    jobid: 13\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/7dpt_chip_rep1_treat_pileup_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/7dpt_chip_rep1_treat_pileup_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWig:\n",
      "    input: bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bigWig\n",
      "    jobid: 26\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgcomp:\n",
      "    input: bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bedGraph, bigWigs/Uninjured_chip_rep2_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep2_subtract.bedGraph\n",
      "    jobid: 78\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "macs2 bdgcmp -t bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bedGraph -c bigWigs/Uninjured_chip_rep2_control_lambda_sort.bedGraph -m subtract -o bigWigs/Uninjured_chip_rep2_subtract.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWigCtl:\n",
      "    input: bigWigs/14dpi_chip_rep1_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep1_control_lambda_sort.bigWig\n",
      "    jobid: 40\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/14dpi_chip_rep1_control_lambda_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/14dpi_chip_rep1_control_lambda_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWig:\n",
      "    input: bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bigWig\n",
      "    jobid: 35\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgcomp:\n",
      "    input: bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bedGraph, bigWigs/Uninjured_chip_rep1_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep1_subtract.bedGraph\n",
      "    jobid: 75\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "macs2 bdgcmp -t bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bedGraph -c bigWigs/Uninjured_chip_rep1_control_lambda_sort.bedGraph -m subtract -o bigWigs/Uninjured_chip_rep1_subtract.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWig:\n",
      "    input: bigWigs/7dpt_chip_rep2_treat_pileup_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep2_treat_pileup_sort.bigWig\n",
      "    jobid: 19\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/7dpt_chip_rep2_treat_pileup_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/7dpt_chip_rep2_treat_pileup_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWigCtl:\n",
      "    input: bigWigs/7dpt_chip_rep1_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep1_control_lambda_sort.bigWig\n",
      "    jobid: 31\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/7dpt_chip_rep1_control_lambda_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/7dpt_chip_rep1_control_lambda_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdg2bigWigCtl:\n",
      "    input: bigWigs/14dpi_chip_rep2_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep2_control_lambda_sort.bigWig\n",
      "    jobid: 18\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/14dpi_chip_rep2_control_lambda_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/14dpi_chip_rep2_control_lambda_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule bdgcomp:\n",
      "    input: bigWigs/7dpt_chip_rep2_treat_pileup_sort.bedGraph, bigWigs/7dpt_chip_rep2_control_lambda_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep2_subtract.bedGraph\n",
      "    jobid: 74\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "macs2 bdgcmp -t bigWigs/7dpt_chip_rep2_treat_pileup_sort.bedGraph -c bigWigs/7dpt_chip_rep2_control_lambda_sort.bedGraph -m subtract -o bigWigs/7dpt_chip_rep2_subtract.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule subSort:\n",
      "    input: bigWigs/14dpi_chip_rep2_subtract.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep2_subtract_sort.bedGraph\n",
      "    jobid: 63\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "bedtools sort -i bigWigs/14dpi_chip_rep2_subtract.bedGraph > bigWigs/14dpi_chip_rep2_subtract_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule subSort:\n",
      "    input: bigWigs/7dpt_chip_rep1_subtract.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep1_subtract_sort.bedGraph\n",
      "    jobid: 69\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "bedtools sort -i bigWigs/7dpt_chip_rep1_subtract.bedGraph > bigWigs/7dpt_chip_rep1_subtract_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule subSort:\n",
      "    input: bigWigs/14dpi_chip_rep1_subtract.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep1_subtract_sort.bedGraph\n",
      "    jobid: 46\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "bedtools sort -i bigWigs/14dpi_chip_rep1_subtract.bedGraph > bigWigs/14dpi_chip_rep1_subtract_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule subSort:\n",
      "    input: bigWigs/Uninjured_chip_rep2_subtract.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep2_subtract_sort.bedGraph\n",
      "    jobid: 71\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "bedtools sort -i bigWigs/Uninjured_chip_rep2_subtract.bedGraph > bigWigs/Uninjured_chip_rep2_subtract_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule subSort:\n",
      "    input: bigWigs/7dpt_chip_rep2_subtract.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep2_subtract_sort.bedGraph\n",
      "    jobid: 48\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "bedtools sort -i bigWigs/7dpt_chip_rep2_subtract.bedGraph > bigWigs/7dpt_chip_rep2_subtract_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule subSort:\n",
      "    input: bigWigs/Uninjured_chip_rep1_subtract.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep1_subtract_sort.bedGraph\n",
      "    jobid: 53\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "bedtools sort -i bigWigs/Uninjured_chip_rep1_subtract.bedGraph > bigWigs/Uninjured_chip_rep1_subtract_sort.bedGraph\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule sub2bigWig:\n",
      "    input: bigWigs/14dpi_chip_rep1_subtract_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep1_subtract_sort.bigWig\n",
      "    jobid: 3\n",
      "    wildcards: sample=14dpi_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/14dpi_chip_rep1_subtract_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/14dpi_chip_rep1_subtract_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule sub2bigWig:\n",
      "    input: bigWigs/7dpt_chip_rep1_subtract_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep1_subtract_sort.bigWig\n",
      "    jobid: 38\n",
      "    wildcards: sample=7dpt_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/7dpt_chip_rep1_subtract_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/7dpt_chip_rep1_subtract_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule sub2bigWig:\n",
      "    input: bigWigs/14dpi_chip_rep2_subtract_sort.bedGraph\n",
      "    output: bigWigs/14dpi_chip_rep2_subtract_sort.bigWig\n",
      "    jobid: 27\n",
      "    wildcards: sample=14dpi_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/14dpi_chip_rep2_subtract_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/14dpi_chip_rep2_subtract_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule sub2bigWig:\n",
      "    input: bigWigs/Uninjured_chip_rep2_subtract_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep2_subtract_sort.bigWig\n",
      "    jobid: 41\n",
      "    wildcards: sample=Uninjured_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/Uninjured_chip_rep2_subtract_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/Uninjured_chip_rep2_subtract_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule sub2bigWig:\n",
      "    input: bigWigs/Uninjured_chip_rep1_subtract_sort.bedGraph\n",
      "    output: bigWigs/Uninjured_chip_rep1_subtract_sort.bigWig\n",
      "    jobid: 14\n",
      "    wildcards: sample=Uninjured_chip_rep1\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/Uninjured_chip_rep1_subtract_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/Uninjured_chip_rep1_subtract_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "rule sub2bigWig:\n",
      "    input: bigWigs/7dpt_chip_rep2_subtract_sort.bedGraph\n",
      "    output: bigWigs/7dpt_chip_rep2_subtract_sort.bigWig\n",
      "    jobid: 8\n",
      "    wildcards: sample=7dpt_chip_rep2\n",
      "\n",
      "/home/vschuma/bin/bedGraphToBigWig bigWigs/7dpt_chip_rep2_subtract_sort.bedGraph https://hgdownload-test.gi.ucsc.edu/goldenPath/currentGenomes/Danio_rerio/bigZips/danRer11.chrom.sizes bigWigs/7dpt_chip_rep2_subtract_sort.bigWig\n",
      "\n",
      "[Fri Sep 20 18:12:21 2019]\n",
      "localrule all:\n",
      "    input: samtools/sammark_mapped_Uninjured_input.bam, samtools/sammark_mapped_Uninjured_chip_rep1.bam, samtools/sammark_mapped_Uninjured_chip_rep2.bam, samtools/sammark_mapped_Uninjured_chip_rep3.bam, samtools/sammark_mapped_14dpi_input.bam, samtools/sammark_mapped_14dpi_chip_rep1.bam, samtools/sammark_mapped_14dpi_chip_rep2.bam, samtools/sammark_mapped_7dpt_input.bam, samtools/sammark_mapped_7dpt_chip_rep1.bam, samtools/sammark_mapped_7dpt_chip_rep2.bam, samtools/sammark_mapped_Uninjured_input.bam.bai, samtools/sammark_mapped_Uninjured_chip_rep1.bam.bai, samtools/sammark_mapped_Uninjured_chip_rep2.bam.bai, samtools/sammark_mapped_Uninjured_chip_rep3.bam.bai, samtools/sammark_mapped_14dpi_input.bam.bai, samtools/sammark_mapped_14dpi_chip_rep1.bam.bai, samtools/sammark_mapped_14dpi_chip_rep2.bam.bai, samtools/sammark_mapped_7dpt_input.bam.bai, samtools/sammark_mapped_7dpt_chip_rep1.bam.bai, samtools/sammark_mapped_7dpt_chip_rep2.bam.bai, macs/Uninjured_chip_rep1_peaks.narrowPeak, macs/Uninjured_chip_rep2_peaks.narrowPeak, macs/14dpi_chip_rep1_peaks.narrowPeak, macs/14dpi_chip_rep2_peaks.narrowPeak, macs/7dpt_chip_rep1_peaks.narrowPeak, macs/7dpt_chip_rep2_peaks.narrowPeak, bigWigs/Uninjured_chip_rep1_treat_pileup_sort.bigWig, bigWigs/Uninjured_chip_rep2_treat_pileup_sort.bigWig, bigWigs/14dpi_chip_rep1_treat_pileup_sort.bigWig, bigWigs/14dpi_chip_rep2_treat_pileup_sort.bigWig, bigWigs/7dpt_chip_rep1_treat_pileup_sort.bigWig, bigWigs/7dpt_chip_rep2_treat_pileup_sort.bigWig, bigWigs/Uninjured_chip_rep1_control_lambda_sort.bigWig, bigWigs/Uninjured_chip_rep2_control_lambda_sort.bigWig, bigWigs/14dpi_chip_rep1_control_lambda_sort.bigWig, bigWigs/14dpi_chip_rep2_control_lambda_sort.bigWig, bigWigs/7dpt_chip_rep1_control_lambda_sort.bigWig, bigWigs/7dpt_chip_rep2_control_lambda_sort.bigWig, bigWigs/Uninjured_chip_rep1_subtract_sort.bigWig, bigWigs/Uninjured_chip_rep2_subtract_sort.bigWig, bigWigs/14dpi_chip_rep1_subtract_sort.bigWig, bigWigs/14dpi_chip_rep2_subtract_sort.bigWig, bigWigs/7dpt_chip_rep1_subtract_sort.bigWig, bigWigs/7dpt_chip_rep2_subtract_sort.bigWig\n",
      "    jobid: 0\n",
      "\n",
      "Job counts:\n",
      "\tcount\tjobs\n",
      "\t1\tall\n",
      "\t6\tbdg2bigWig\n",
      "\t6\tbdg2bigWigCtl\n",
      "\t6\tbdgSort\n",
      "\t6\tbdgSortCtl\n",
      "\t6\tbdgcomp\n",
      "\t6\tsub2bigWig\n",
      "\t6\tsubSort\n",
      "\t43\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "snakemake -p -s snakemake_goldmanH33_data.py --directory ../results -n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#qsub file\n",
    "cat>qsub_goldmanH33_data.sh<<EOL\n",
    "#!/bin/bash\n",
    "#$ -N goldman_data    # name of the job\n",
    "#$ -m ea                   # send an email when end or job aborted \n",
    "#$ -M Vic-Fabienne.Schumann@mdc-berlin.de \n",
    "#$ -wd   /fast/AG_Ohler/Vic/0_Projects/Goldman_injured/src    # working dir\n",
    "#$ -j y                        # combine error with stdout\n",
    "#$ -o log_error.stdlog      # default filename for stdout and error file \n",
    "##$ -l h_rt=10:00:00\n",
    "#$ -l m_mem_free=40G    # limit memory request\n",
    "#$ -l h_stack=128M    # often not needed, but a few programs require it\n",
    "#$ -pe smp 4            # smp parallel environment\n",
    "\n",
    "\n",
    "\n",
    "# I have this in every qsub file, so that I know what the job was, what node it was on if there are problems, and the resources I requested\n",
    "\n",
    "echo \"===================\"\n",
    "date\n",
    "echo \"job: \"\n",
    "echo \"jobID: \"\n",
    "echo \"node: \"max162.mdc-berlin.net\n",
    "/opt/uge/bin/lx-amd64/qstat -f -j  | grep\n",
    "\"job_number\\|usage\\|resource_list\\|job_args\\|stdout_path_list\"\n",
    "echo \"===================\"\n",
    "\n",
    "\n",
    "\n",
    "# BODY: run the program of interest \n",
    "source /home/vschuma/.guix-profile/etc/profile\n",
    "\n",
    "snakemake -p -s snakemake_goldmanH33_data.py --directory ../results -j 4 \n",
    "\n",
    "# at the bottom of every qsub file I have the below so that I know the resources that were used for that particular job. I do get this in the final email, but sometimes it's nicer to have it searchable in the log file.\n",
    "\n",
    "/opt/uge/bin/lx-amd64/qstat -f -j  | grep \"usage\\|resource_list\"\n",
    "EOL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.10.2019\n",
    "\n",
    "Preparation steps for DeSeq analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writefile snakemake_prep_DeSeq_Goldman_ChIP.py\n",
    "\n",
    "configfile: '../src/config_goldmanH33_data.yaml'\n",
    "\n",
    "FILE_NAME  = config[\"bigWigs\"]\n",
    "\n",
    "merged_sample_names = []\n",
    "for samplename in FILE_NAME:\n",
    "    if 'wt' in samplename:\n",
    "        if 'rep1' in samplename:\n",
    "            merged_sample_names.append(samplename[:-5])\n",
    "    \n",
    "\n",
    "\n",
    "rule all:\n",
    "    input:\n",
    "#bedtools\n",
    "        \"merged/merged_injured_U7dpi14dpi.bed\",\n",
    "        \"matrix/deseq_matrix_injured_U7dpi14dpi.bam\"\n",
    "    \n",
    "include: \"rules/bedtools_merge_for_deseq.smk\"\n",
    "include: \"rules/multicov_for_deseq.smk\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "/gnu/store/3w3nz0h93h7jif9d9c3hdfyimgkpx1a4-python-wrapper-3.7.0/bin/python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
